src/coreclr/vm/mono/mono_coreclr.cpp

```diff
@@ -3091,7 +3141,6 @@ extern "C" EXPORT_API void EXPORT_CC mono_unity_thread_fast_attach (MonoDomain *
 extern "C" EXPORT_API void EXPORT_CC mono_unity_thread_fast_detach ()
 {
     gCurrentDomain = NULL;
-    GetThread()->EnablePreemptiveGC();
```

Comment:
I'm not sure about the usecase here, but it is best to not change the thread mode as subsequent managed api calls on job thread cause runtime throw mode exceptions

Reply:
Agreed, I think this API doesn't have same meaning or implications for CoreCLR as Mono.